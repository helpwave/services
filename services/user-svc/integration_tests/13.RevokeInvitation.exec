#!/bin/bash

echo "exists:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6ImhlaW5yaWNoLmhlaW5lQGhlbHB3YXZlLmRlIiwibmFtZSI6IlRlc3RpbmUgVGVzdCIsIm5pY2tuYW1lIjoidGVzdGluZS50ZXN0Iiwib3JnYW5pemF0aW9ucyI6WyIzYjI1YzZmNS00NzA1LTQwNzQtOWZjNi1hNTBjMjhlYmE0MDYiXX0=" \
	-d '{"invitation_id": "1bb75dca-0ec0-485e-85f4-b7a813c89a6b" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/RevokeInvitation

echo "exists, not pending"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6ImhlaW5yaWNoLmhlaW5lQGhlbHB3YXZlLmRlIiwibmFtZSI6IlRlc3RpbmUgVGVzdCIsIm5pY2tuYW1lIjoidGVzdGluZS50ZXN0Iiwib3JnYW5pemF0aW9ucyI6WyIzYjI1YzZmNS00NzA1LTQwNzQtOWZjNi1hNTBjMjhlYmE0MDYiXX0=" \
	-d '{"invitation_id": "6acc312b-62ce-4077-ab74-2f6794272b9c" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/RevokeInvitation 2>&1 || echo ""

echo "does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6Im1haWtlLnRlc3RAaGVscHdhdmUuZGUiLCJuYW1lIjoiVGVzdGluZSBUZXN0Iiwibmlja25hbWUiOiJ0ZXN0aW5lLnRlc3QiLCJvcmdhbml6YXRpb25zIjpbIjNiMjVjNmY1LTQ3MDUtNDA3NC05ZmM2LWE1MGMyOGViYTQwNiJdfQ==" \
	-d '{"invitation_id": "1c9b2374-3815-4529-876f-217b32f0303e" }' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/RevokeInvitation 2>&1 || echo ""
