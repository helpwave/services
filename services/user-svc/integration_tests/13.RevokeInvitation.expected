exists:
{}
exists, not pending
ERROR:
  Code: InvalidArgument
  Message: only pending invitations can be revoked

does not exist:
ERROR:
  Code: InvalidArgument
  Message: record not found

